sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -481,10 +482,10 @@ case class Cast(
 
   override def nullable: Boolean = child.nullable || Cast.forceNullable(child.dataType, dataType)
 
-  override def initQueryContext(): String = if (ansiEnabled) {
-    origin.context
+  override def initQueryContext(): Option[QueryContext] = if (ansiEnabled) {
```

Comment:
maybe we can use `SQLQueryContext` everywhere in SQL, to carry the summary string.

Reply:
done